master/internal/rm/kubernetesrm/kubernetes_resource_manager.go

```diff
@@ -433,7 +432,7 @@ func (k ResourceManager) ResolveResourcePool(
 	}
 	found := false
 	for _, poolName := range poolNames {
-		if name == poolName {
+		if string(name) == poolName {
```

Comment:
What do you think about defining a `String()` method for the `ResourcePoolName` type?

Reply:
I'm thinking that it would help separate the cases where we're creating a new `ResourcePoolName` using `string( < text > )` and accessing the value of `ResourcePoolName` using `resourcePoolName.String()`.

Reply:
i like this. i do it in most cases. i know `string(name)` works just fine, but if you do `type ResourcePoolName string` and define `String()` on it, then refactoring ResourcePoolName to not be a string is easier.

but, i mean, it's easy either way, so that isn't a killer argument.